loader/src/main/java/net/neoforged/fml/loading/moddiscovery/locators/JarInJarDependencyLocator.java

```diff
@@ -109,9 +109,14 @@ private String formatError(final ModWithVersionRange modWithVersionRange) {
     }
 
     protected String identifyMod(final IModFile modFile) {
-        if (modFile.getModFileInfo() == null || modFile.getModInfos().isEmpty()) {
+        if (modFile.getModFileInfo() == null) {
             return modFile.getFileName();
         }
+        // If this is a library, it won't have any mod IDs, so we use the module name instead.
+        if (modFile.getModInfos().isEmpty()) {
+            // Prefix to ensure this cannot collide with any true mod ID.
+            return "library:" + modFile.getModFileInfo().moduleName();
+        }
 
         return modFile.getModInfos().stream().map(IModInfo::getModId).collect(Collectors.joining());
```

Comment:
If we're already fixing up this logic, can we fix this logic here too, by joining them with something that _can't_ be in a mod ID, like `+` or `,`? The current logic would treat a jar containing the mods `foo` and `bar` as identical to a jar containing the mod `foobar`, which it is not.

Reply:
out of scope IMO

Reply:
The other option is to take just the _first_ mod ID -- which may be more sensible, as that is what becomes the module name. That said -- is there a reason we don't just use the module name in _general_ here? After all, it'll be unique for different mods and for other stuff, but the same between multiple mods that are the "same" mod or whatever.